lib/cdo/test_run_utils.rb

```
@@ -26,10 +26,110 @@ def self.run_blockly_core_tests
     end
   end
 
-  def self.run_dashboard_tests(parallel: false)
+  # Setup method to prepare test databases for parallel dashboard tests,
+  # which can be called from the QA pass (the test machine) or CI (Drone).
+  def self.setup_dashboard_tests_parallel(upload_seed_data: false)
+    # Parallel tests don't seem to run more quickly over 16 processes.
+    ENV['PARALLEL_TEST_PROCESSORS'] = '16' if RakeUtils.nproc > 16
```

This would only happen if we've configured `PARALLEL_TEST_PROCESSORS` too high in the "prepare_ci_tests.sh" script, right? Should we warn or fail if we've configured it too high, rather than silently capping it?

---

that's the only time it could happen within CI. However, this does happen in practice on the Test machine where we do not specify PARALLEL_TEST_PROCESSORS and the number of CPUs available is greater than 16:
```
[test] dashboard > require 'cdo/rake_utils'
[test] dashboard > RakeUtils.nproc
=> 96
```
Most of this code was moved unmodified from `dashboard_ci`/`dashboard_qa` in test.rake. you can see my changes to it by viewing this commit range: https://github.com/code-dot-org/code-dot-org/pull/69155/files/36700a09beffb429ae0b4e11d6fb9758c862e935..8183676990c4f5dc0bfff852360924fc499621c1